hotsos/core/host_helpers/cli.py

```diff
@@ -129,18 +152,16 @@ class FileCmd(CmdBase):
 
     def __init__(self, path, json_decode=False,
                  singleline=False, decode_error_handling=None):
-        self.path = self.original_path = os.path.join(HotSOSConfig.data_root,
-                                                      path)
-        self.original_json_decode = json_decode
-        self.original_singleline = singleline
-        self.original_decode_error_handling = decode_error_handling
+        self.state = FileCmdState(os.path.join(HotSOSConfig.data_root,
+                                               path), json_decode, singleline,
+                                  decode_error_handling)
+        self.reset()
```

Comment:
I think this code became unnecessary too complicated.
Thinking out lout, when you crate object of FileCmdState at the end of object creation you want to have values to be set on both path, json_decode (...etc) and orginal_path, original_json_decode (...etc)
Why not to call FileCmdState.reset in FileCmdState._post_init_?

Next, going to line 158, you are calling self.reset() to copy all attributes from self.state (FileCmdState) to self._attribute_name(FileCmd), is that necessary? You already have self.state.path why also add self.path?

Reply:
I'm not saying code won't work, but it's getting a little bit confusing.

Reply:
i have changed the way this works now, and hopefully made it more intuitive. please take a look